src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.epics.js

```diff
@@ -42,12 +43,24 @@ const getFiltersForAttributesSearchQuery = formValues =>
         .map(fieldId => `${fieldId}:like:${formValues[fieldId]}`);
 
 
-const searchViaAttributesStream = (queryArgs, attributes) =>
+const searchViaAttributesStream = (queryArgs, attributes, triggeredFrom) =>
     from(getTrackedEntityInstances(queryArgs, attributes)).pipe(
         map(({ trackedEntityInstanceContainers: searchResults, pagingData }) => {
             if (searchResults.length > 0) {
-                return showSuccessResultsViewOnSearchPage(searchResults, pagingData);
+                const nextPageButtonDisabled = Boolean(searchResults.length < pagingData.rowsPerPage);
+                return showSuccessResultsViewOnSearchPage(
+                    searchResults,
+                    { currentPage: pagingData.currentPage, nextPageButtonDisabled },
+                );
             }
+
+            if (searchResults.length === 0 && triggeredFrom === PAGINATION) {
+                return showSuccessResultsViewOnSearchPage(
+                    [],
+                    { currentPage: pagingData.currentPage, nextPageButtonDisabled: true },
```

Comment:
One thing I've learned is that it is better to store only what is needed, i.e. save a common state that can be derived from and do the computations when rendering. The `nextPageButtonDisabled` could be derived from state you already have. 

In the same vein, you could derive from the currentPage if you would like to show an empty page or the messages.

Furthermore, the above leads to why I don't like the search status. Is it really needed or could the search status be derived from other state? Is it just going to be a pain to keep the search status up to date and prevent it from going stale?

This is just friendly advice for the future ;) I would appreciate it if we could compute the nextPageButtonDisabled instead of storing it though (lets keep the search status and the other solutions for now)

Reply:
Hey I agree on that you dont wanna be storing _all_ the information and things could derive. 

The state approach I am taking in my eyes ( which are of course biased in this case )  helps the reader understand what to expect from the `SearchPage.component.js` when they start reading it. 

I am removing the `nextPageButtonDisabled` from the state. Can we please not be absolute on the `seachStatus` and give it a go to see how this works in the longer run?

Reply:
@JoakimSM also seeing the code again a little bit. And I want to stress out that we mostly have the same opinion on the topic here. I also strongly agree that things that shouldnt be stored on redux shouldnt be there. However I am just gonna raise another point because I rushed into answering the message before and hadnt though through it well... 

The next button derives from two values at the moment. For example
```
const nextPageButtonDisabled = Boolean(searchResults.length < pagingData.rowsPerPage);
```
the values that used to derive the state of the button are, 
a) the length of the results on the response and 
b) the `rowsPerPage` from the response as well.

Now I have stopped storing the `rowsPerPage` in the reducers. (exception to this is the working reducer list coz there is needed).

So we can store the rowsPerPage and then derive the status of the button but then from this point of view isnt the `rowsPerPage` that isnt really needed since its only used to derive the button state?

Reply:
Or do you mean there is another way to derive the button state that I havent considered?

Reply:
Since we don't allow for changing rows per page, we should just set this in the component itself and pass it as a argument when initiating the search.

Reply:
Which component?

Reply:
> Can we please not be absolute on the seachStatus and give it a go to see how this works in the longer run?

Yes, we're not changing this now. Regarding the readability, you could compute the status in the component.

Reply:
> Regarding the readability, you could compute the status in the component.

Yeah I am happy to understand your point of view on this one and perhaps find a way that both of us like to go forward

Reply:
> Which component?
In the lowest common ancestor to where rows per page is needed :)

Reply:
I still dont think I get that fully. 

At the moment the `pageSize` is hardcoded to number 5. And this is happening usually in the epics. 
The places that this happens at the moment is: 
1. `searchGroupDuplicate.epics`
2. `searchForm.epics`
3. `teiSearch.epics`

What I understand _( and perhaps I am wrong )_ is that now, we would like the number of the pages to be defined on the component level. We could then pass the number alongside with other information in the action creator and then into the epics. This way we can pass _also_ the number 5 into the `Pagination` component that is defined on those 3 places defined above?